### PR TITLE
Endpoint POST /users created

### PR DESCRIPTION
**Pull request by danielWolox**
## Summary

Endpoint POST /users to create an user with it's required validations

## Known Issues

None

## Trello Card

https://trello.com/c/qrAyr2vu/35-sign-up

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const UsersController = module.exports;
+
+const { validationResult } = require('express-validator');
+
+const UsersService = require('../services/users');
```
Comment (marioportela1993) [inline app/controllers/users.js]: We name the variables in camelCase no PascalCase, please check this convention in all parts in the project

File: app/middlewares/validatonSchemas/users.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+const { user: UserModel } = require('../../models/index');
+
+const UserSchemas = module.exports;
+
+const providedAndNotEmpty = 'should be provided and not empty';
```
Comment (marioportela1993) [inline app/middlewares/validatonSchemas/users.js]: Put this type of variables into a file named constants :)

File: app/middlewares/validatonSchemas/users.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+const { user: UserModel } = require('../../models/index');
```
Comment (marioportela1993) [inline app/middlewares/validatonSchemas/users.js]: It is not necessary to indicate the index file. You can do the request like : `const { user: UserModel } = require('../../models');` due to for default the require find the index file.

File: app/middlewares/validatonSchemas/users.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    notEmpty: true,
+    errorMessage: providedAndNotEmpty
+  },
+  email: {
+    in: 'body',
+    optional: false,
+    notEmpty: true,
+    errorMessage: providedAndNotEmpty,
+    isEmail: {
+      errorMessage: 'should be a valid email',
```
Comment (marioportela1993) [inline app/middlewares/validatonSchemas/users.js]: Same comment, put this messages into a constants file or maybe into the errors.js file. I prefer a constants file. What do you think?

File: app/middlewares/validatonSchemas/users.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      bail: true
+    },
+    matches: {
+      options: /(\w|\.|-)+@wolox\.(\w{2}|com)$/i,
+      errorMessage: 'should belong to wolox domain',
+      bail: true
+    },
+    custom: {
+      options: email =>
+        UserModel.findOne({ where: { email } }).then(user => {
```
Comment (marioportela1993) [inline app/middlewares/validatonSchemas/users.js]: Nice!

Comment (marioportela1993) [reply app/middlewares/validatonSchemas/users.js]: What happen if the db is down ?

File: app/models/user.js [hunk 6]
```diff
@@ -10,7 +13,16 @@ module.exports = (sequelize, DataTypes) => {
     },
     {
       tableName: 'users',
-      underscored: true
+      underscored: true,
+      hooks: {
```
Comment (marioportela1993) [inline app/models/user.js:17]: Nice :)

**Review by marioportela1993 — CHANGES_REQUESTED** [review]
Excellent job Dani, check comments and let me know if you have questions

File: app/middlewares/validatonSchemas/users.js (hunk 4 above)
Comment (danielWolox) [reply app/middlewares/validatonSchemas/users.js]: It's also better for me in a constants file

File: app/middlewares/validatonSchemas/users.js (hunk 5 above)
Comment (danielWolox) [reply app/middlewares/validatonSchemas/users.js]: if the database is down, the service would throw a SequelizeConnectionError when we try to start the server, if the database is up but there is a database error in this validation we would throw a badRequestError with message "email database error".

File: app/middlewares/validateSchema.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+const { validationResult } = require('express-validator');
+
+const errors = require('../errors');
+
+module.exports = (req, res, next) => {
```
Comment (marioportela1993) [inline app/middlewares/validateSchema.js]: `exports.validation`

File: app/middlewares/validateSchema.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+  const validationErrors = validationResult(req).array();
+
+  if (validationErrors.length) {
+    return next(errors.badRequestError(`${validationErrors[0].param} ${validationErrors[0].msg}`));
```
Comment (marioportela1993) [inline app/middlewares/validateSchema.js]: `schemaError`

File: app/middlewares/validationSchemas/users.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
+const { user: UserModel } = require('../../models');
+const { errorMessages, regExp } = require('../../constants');
+
+const UserSchemas = module.exports;
+
+UserSchemas.createUserSchema = {
+  first_name: {
+    in: 'body',
```
Comment (marioportela1993) [inline app/middlewares/validationSchemas/users.js:6]: in: ['body', 'query', 'params'] dato random

**Review by marioportela1993 — APPROVED** [review]
Please fix the changes suggested before to merge the PR.

**Pull request merged**